scripts/publish-ci.ts

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env zx
+
+import { readFileSync } from 'node:fs'
+import { fileURLToPath } from 'node:url'
+import { $ } from 'zx'
+
+let version = process.argv[2]
+
+if (!version)
+  throw new Error('No tag specified')
+
+if (version.startsWith('v'))
+  version = version.slice(1)
+
+const pkgPath = fileURLToPath(new URL('../package.json', import.meta.url))
+const pkg = JSON.parse(readFileSync(pkgPath, 'utf-8'))
+
+if (pkg.version !== version) {
+  throw new Error(
+    `Package version from tag "${version}" mismatches with the current version "${pkg.version}"`,
+  )
+}
+
+const releaseTag = version.includes('beta')
+  ? 'beta'
+  : version.includes('alpha')
+    ? 'alpha'
+    : undefined
+
+console.log('Publishing version', version, 'with tag', releaseTag)
+
+if (releaseTag)
+  await $`pnpm -r publish --access public --tag ${releaseTag}`
+else
+  await $`pnpm -r publish --access public`
```

Comment:
Does the `-r` now include all packages mentioned in `pnpm-workspace.yaml`?

https://github.com/vitest-dev/vitest/blob/e4e93dfd04dd6d680d5dc1431cd57f4c2c1f210e/pnpm-workspace.yaml#L1-L6

Should we instead add a `--filter` flag here to target only `vitest` and `@vitest/*` packages?

Reply:
This is the same command we use right now, why would it change?